vm/loader/loader_defs/src/shim.rs

```diff
@@ -88,6 +88,8 @@ open_enum! {
         VTL0_MMIO = 5,
         /// This range is mmio for VTL2.
         VTL2_MMIO = 6,
+        /// Memory preserved during servicing.
+        VTL2_PRESERVED = 7,
```

Comment:
How do these entries make it into the tables? Does this mean we tell the host about these ranges somehow?

Reply:
Or does the host just know what the preserved _range_ is? Is this some new IGVM concept?

Reply:
Host knows what the DMA preserved size should be but does not calculate the ranges. Host provides size through device tree, boot shim selects the top of vtl2 memory range and marks it as reserved with the new type.

Reply:
I think we need some way to enforce that the next version of ohcl selects the same size as previous. Outside the scope of this PR, but i'll incorporate it in some changes I'm making moving forward.

Reply:
There is future update in boot_shim + PagePool allocator